Documentation/Kernel/DevelopmentGuidelines.md

```diff
@@ -198,6 +198,31 @@ We allocate them in `Kernel/API/MajorNumberAllocation.h`, based on this set of r
 6. The allocation must be inserted by keeping the ascending order of the allocated major numbers
    in the appropriate enum and associated to-StringView function.
 
+## Construction of `Device`-based objects
+
+Currently, we have many devices that are either inserted at boot time but also devices that could be inserted
+afterwards.
+
+To make it easier writing device drivers, when constructing an object from a `Device`-derived class, the usual pattern is to use `DeviceManagement` `try_create_device` method.
+For example, constructing a `VirtIOGPU3DDevice` is done this way:
+```c++
+ErrorOr<NonnullLockRefPtr<VirtIOGPU3DDevice>> VirtIOGPU3DDevice::try_create(VirtIOGraphicsAdapter& adapter)
+{
+    // Setup memory transfer region
```

Comment:
Oh, and this is `Set up memory ...`. "Setup" is the noun, "to set up" is the verb.

Reply:
This was taken from the actual code, but I guess it's OK to change this here :shrug: